GitFyle.Core.Api.Tests.Unit/Services/Foundations/Configurations/ConfigurationServiceTests.Validations.Add.cs

```diff
@@ -0,0 +1,330 @@
+                key: nameof(Configuration.UpdatedBy),
+                values: $"Text is not same as {nameof(Configuration.CreatedBy)}");
+
+            invalidConfigurationException.AddData(
+                key: nameof(Configuration.UpdatedDate),
+                values: $"Date is not same as {nameof(Configuration.CreatedDate)}");
+
+            var expectedConfigurationValidationException =
+                new ConfigurationValidationException(
+                    message: "Configuration validation error occurred, fix the errors and try again.",
+                    innerException: invalidConfigurationException);
+
+            this.datetimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffsetAsync())
+                    .ReturnsAsync(now);
+
+            // when
+            ValueTask<Configuration> addConfigurationTask =
+                this.configurationService.AddConfigurationAsync(invalidConfiguration);
+
+            ConfigurationValidationException actualConfigurationValidationException =
+                await Assert.ThrowsAsync<ConfigurationValidationException>(
+                    addConfigurationTask.AsTask);
+
+            // then
+            actualConfigurationValidationException.Should()
+                .BeEquivalentTo(expectedConfigurationValidationException);
+
+            this.datetimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffsetAsync(),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogErrorAsync(It.Is(
+                    SameExceptionAs(expectedConfigurationValidationException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.InsertConfigurationAsync(It.IsAny<Configuration>()),
+                    Times.Never);
+
+            this.datetimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(-61)]
+        public async Task ShouldThrowValidationExceptionOnAddIfCreatedDateIsNotRecentAndLogItAsync(int invalidSeconds)
+        {
+            // given
+            DateTimeOffset randomDateTime = 
+                GetRandomDateTimeOffset();
+
+            DateTimeOffset now = randomDateTime;
+            DateTimeOffset startDate = now.AddSeconds(-60);
+            DateTimeOffset endDate = now.AddSeconds(0);
+            Configuration randomConfiguration = CreateRandomConfiguration();
+            Configuration invalidConfiguration = randomConfiguration;
+
+            DateTimeOffset invalidDate =
+                now.AddSeconds(invalidSeconds);
+
+            invalidConfiguration.CreatedDate = invalidDate;
+            invalidConfiguration.UpdatedDate = invalidDate;
+
+            var invalidConfigurationException = new InvalidConfigurationException(
```

Comment:
Format this the same as well.

Reply:
Fixed.

![image](https://github.com/user-attachments/assets/8f5f0a76-6e2e-4b89-9851-f588f5abbcb3)